engine/api/src/main/scala/pl/touk/nussknacker/engine/api/context/UsageClosure.scala

```diff
@@ -0,0 +1,23 @@
+package pl.touk.nussknacker.engine.api.context
+
+import pl.touk.nussknacker.engine.api.typed.typing.{TypedClass, TypingResult}
+
+sealed trait UsageClosure
+
+case class VariableName(name: String) extends UsageClosure
+
+// TODO: should be Expression in values?
+case class BranchExpressions(expresionByBranchId: Map[String, ValidationContext => TypedClass]) extends UsageClosure
+
+case class CompileTimeEvaluatedParameter[T](name: String, value: T) extends UsageClosure
+
+
+sealed trait UsageClosureDefinition
+
+case object VariableNameDef extends UsageClosureDefinition
+
+case object BranchExpressionsDef extends UsageClosureDefinition
+
+case class CompileTimeEvaluatedParameterDef(name: String, typ: TypingResult)
```

Comment:
in some cases TypingResult will be Unknown, and the *real* result can be given only by applying context...